runtime/moonbase/src/precompiles.rs

```diff
@@ -70,11 +71,17 @@ impl<R> PrecompileSet for MoonbasePrecompiles<R>
 where
 	R::Call: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo + Decode,
 	<R::Call as Dispatchable>::Origin: From<Option<R::AccountId>>,
-	R: parachain_staking::Config + pallet_evm::Config + pallet_crowdloan_rewards::Config,
-	R::AccountId: From<H160>,
+	R: parachain_staking::Config
+		+ pallet_evm::Config
+		+ pallet_crowdloan_rewards::Config
+		+ pallet_democracy::Config,
+	R::AccountId: From<H160>, //TODO we shouldn't need this. There is a method for it in frontier.
 	BalanceOf<R>: Debug + precompile_utils::EvmData,
 	RewardBalanceOf<R>: TryFrom<sp_core::U256> + Debug,
-	R::Call: From<parachain_staking::Call<R>> + From<pallet_crowdloan_rewards::Call<R>>,
+	R::Call: From<parachain_staking::Call<R>>
+		+ From<pallet_crowdloan_rewards::Call<R>>
+		+ From<pallet_democracy::Call<R>>,
+	DemocracyWrapper<R>: Precompile, //TODO why is this necessary? And why only for democracy?
```

Comment:
You shouldn't have to do this. I remember having the Rust compiler speaking about this missing bound, but it was the side effect of another issue.

Reply:
I also expected that it was a symptom of another issue. But when I added it, it compiles successfully.

Reply:
I worked on this more in 84964ed but still can't get it to compile without `DemocracyWrapper<R>: Precompile`.

Reply:
Turns out this was because my `Precoompile` had defined `BalanceOf<T>` in terms of pallet democracy whereas the `PrecompileSet` had defined it in terms of pallet balances. We've decided to address this by not repeating trait bounds on the set, but rather using the `Precompile` bound.